File: cs/src/Ice/Object_Del.cs
// **********************************************************************
//
// Copyright (c) 2003
// ZeroC, Inc.
// Billerica, MA, USA
//
// All Rights Reserved.
//
// Ice is free software; you can redistribute it and/or modify it under
// the terms of the GNU General Public License version 2 as published by
// the Free Software Foundation.
//
// **********************************************************************
namespace Ice
{
	
    public interface Object_Del
    {
	bool ice_isA(string id, Ice.Context context);
	void  ice_ping(Ice.Context context);
	StringSeq ice_ids(Ice.Context context);
	string ice_id(Ice.Context context);
	FacetPath ice_facets(Ice.Context context);
	bool ice_invoke(string operation, Ice.OperationMode mode, Ice.ByteSeq inParams, out Ice.ByteSeq outParams,
			Ice.Context context);
	void ice_invoke_async(AMI_Object_ice_invoke cb, string operation, Ice.OperationMode mode, Ice.ByteSeq inParams,
			      Ice.Context context);
    }

}

File: java/src/Ice/_ObjectDel.java
// **********************************************************************
//
// Copyright (c) 2003-2013 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

package Ice;

import Ice.Instrumentation.InvocationObserver;

public interface _ObjectDel
{
    boolean ice_isA(String id, java.util.Map<String, String> context, InvocationObserver obsv)
        throws IceInternal.LocalExceptionWrapper;

    void ice_ping(java.util.Map<String, String> context, InvocationObserver obsv)
        throws IceInternal.LocalExceptionWrapper;

    String[] ice_ids(java.util.Map<String, String> context, InvocationObserver obsv)
        throws IceInternal.LocalExceptionWrapper;

    String ice_id(java.util.Map<String, String> context, InvocationObserver obsv)
        throws IceInternal.LocalExceptionWrapper;

    boolean ice_invoke(String operation, Ice.OperationMode mode, byte[] inParams, ByteSeqHolder outParams,
                       java.util.Map<String, String> context, InvocationObserver obsv)
        throws IceInternal.LocalExceptionWrapper;

    void ice_flushBatchRequests(InvocationObserver obsv);
    
    IceInternal.RequestHandler __getRequestHandler();
    void __setRequestHandler(IceInternal.RequestHandler handler);
}
